homework-submission/todo-app/src/App.js

```diff
@@ -0,0 +1,69 @@
+import React, { Component } from "react";
+import "./App.css";
+import Item from "./components/Item";
+import ItemData from "./components/ItemData";
+import MainForm from "./components/MainForm";
+
+class App extends Component {
+  constructor() {
+    super();
+    this.state = {
+      todos: ItemData
+    };
+  }
+
+  handleCompleted = index => {
+    const todos = [...this.state.todos];
+    todos[index].done = !todos[index].done;
+    this.setState({ todos });
+  };
+
+  handleSubmit = newItem => {
+    const oldState = this.state.todos;
+    const ItemData = oldState.push(newItem);
+    this.setState({ ItemData });
+  };
+
+  handleRemove = index => {
+    const todos = [...this.state.todos];
+    todos.splice(index, 1);
+    this.setState({ todos });
+  };
+
+  handleUpdate = (newItem, index) => {
+    this.setState(prevState => {
+      prevState.todos[index] = newItem;
+      return prevState;
+    });
+  };
+
+  render() {
+    const data = this.state.todos.map((todo, key) => (
+      <Item
+        todo={todo}
+        key={key}
+        index={key}
+        onCompleted={this.handleCompleted}
+        onEdit={this.handleEdit}
+        onRemove={this.handleRemove}
+        onCancel={this.handleCancel}
```

Comment:
not needed 
??????

Reply:
it gives error if commented out (TypeError: _this.props.onUpdate is not a function) in Item.js:9

Reply:
i am talking about : 
  onCancel={this.handleCancel}